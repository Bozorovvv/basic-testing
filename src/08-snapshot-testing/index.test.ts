import { generateLinkedList } from '08-snapshot-testing';

describe('generateLinkedList', () => {
  test('generates linked list from a non-empty array of numbers', () => {
    const input = [1, 2, 3];
    const expectedOutput = {
      value: 1,
      next: {
        value: 2,
        next: {
          value: 3,
          next: {
            value: null,
            next: null,
          },
        },
      },
    };

    const result = generateLinkedList(input);
    expect(result).toStrictEqual(expectedOutput);
  });

  test('matches the linked list structure for strings with snapshot', () => {
    const input = ['a', 'b', 'c'];
    const result = generateLinkedList(input);

    expect(result).toMatchSnapshot();
  });

  test('returns an empty linked list for an empty array', () => {
    const input: any[] = [];
    const expectedOutput = { value: null, next: null };

    const result = generateLinkedList(input);
    expect(result).toStrictEqual(expectedOutput);
  });
});
