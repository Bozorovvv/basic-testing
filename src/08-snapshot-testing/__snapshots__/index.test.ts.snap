// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList matches the linked list structure for strings with snapshot 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "c",
    },
    "value": "b",
  },
  "value": "a",
}
`;
